test/simpleUpgrader.js

```diff
@@ -27,11 +27,19 @@ describe("Test SimpleUpgrader", function () {
 
         it("should register modules in the registry", async () => {
             let name = "test_1.1";
-            let module = await deployer.deploy(Module, {}, registry.contractAddress, false, 0);
-            await registry.registerModule(module.contractAddress, ethers.utils.formatBytes32String(name));
-            let isRegistered = await registry.isRegisteredModule(module.contractAddress);
+            let initialModule = await deployer.deploy(Module, {}, registry.contractAddress, false, 0);
+            await registry.registerModule(initialModule.contractAddress, ethers.utils.formatBytes32String(name));
+            let isRegistered;
+            // Here we adjust how we call isRegisteredModule which has 2 overlaods, one accepting a single address
+            // and a second accepting an array of addresses. Behaviour as to which overload is selected to run
+            // differs between CI and Coverage environments, adjusted for this here
+            if (process.env.CI && process.env.SOLIDITY_COVERAGE) {
```

Comment:
The first branch of this `if` needs to be executed whenever `SOLIDITY_COVERAGE=1`, instead of when `CI=1 && SOLIDITY_COVERAGE=1`, otherwise `npm run test:coverage` fails on this test when ran locally (when `CI=0`).

Reply:
I'm not sure what you mean. This test passes both in CI and locally and in `test` and in `test:coverage`. How do I replicate the problem you are talking about?

Reply:
Running `npm run test:coverage` fails with:
```
348 passing (13m)
  1 failing

  1) Test SimpleUpgrader
       Registering modules
         should register modules in the registry:
     Error: invalid value for array
      at resolveAddresses (node_modules/etherlime-lib/node_modules/ethers/contract.js:92:19)
      at /Users/ovdbigge/argent/argcover/node_modules/etherlime-lib/node_modules/ethers/contract.js:78:29
      at Array.forEach (<anonymous>)
      at resolveAddresses (node_modules/etherlime-lib/node_modules/ethers/contract.js:70:19)
      at EtherlimeGanacheWrapper.isRegisteredModule (node_modules/etherlime-lib/node_modules/ethers/contract.js:141:16)
      at Context.it (.coverage_tests/simpleUpgrader.js:39:47)
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:189:7)

Error: Some of the test scenarios failed!
    at Object.runCoverage (/Users/ovdbigge/argent/argcover/node_modules/etherlime/cli-commands/etherlime-test/etherlime-coverage.js:84:9)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
[...]
TypeError: Cannot read property 'stop' of undefined
    at Object.commandProcessor [as handler] (/Users/ovdbigge/argent/argcover/node_modules/etherlime/cli-commands/commands.js:421:21)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! argent-contracts@1.0.0 test:coverage: `bash ./scripts/provision_lib_artefacts.sh & SOLIDITY_COVERAGE=1 npx etherlime coverage --solcVersion 0.5.4 && istanbul check-coverage --statements 74 --branches 60 --functions 77 --lines 74`
npm ERR! Exit status 1
[...]
```

Replacing L36 with `if (process.env.SOLIDITY_COVERAGE) {` fixes the issue.

Reply:
It runs okay for me locally :/

Reply:
Weird :( You don't have a local environment variable `CI` set to `1` by any chance?

Reply:
```
process.env.CI undefined
process.env.SOLIDITY_COVERAGE 1
```
Have you switched to node 10?

Reply:
@juniset just ran coverage okay as well. Definitely something with your environment.

Reply:
Yes I'm using node 10.12.

What is the reason behind checking for `process.env.CI == 1` in the first place btw? Why isn't it in principle sufficient to check for `process.env.SOLIDITY_COVERAGE` ?

Reply:
It's to work around a possible bug in either `ethers` or `etherlime` which picks up the wrong overload for `isRegisteredModule` and fails in CI, see [sample failing build](https://circleci.com/gh/argentlabs/argent-contracts/229).